ravedude/src/config.rs

```diff
@@ -0,0 +1,181 @@
+use serde::{Deserialize, Serialize};
+use std::num::NonZeroU32;
+
+#[derive(serde::Serialize, serde::Deserialize, Debug, Default)]
+#[serde(rename_all = "kebab-case")]
+pub struct RavedudeConfig {
+    #[serde(rename = "general")]
+    pub general_options: RavedudeGeneralConfig,
+
+    #[serde(rename = "board")]
+    pub board_config: Option<BoardConfig>,
+}
+
+fn serialize_baudrate<S>(val: &Option<Option<NonZeroU32>>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    let baudrate = val.as_ref().map(|val| val.map_or(-1, |x| x.get() as i32));
+
+    baudrate.serialize(serializer)
+}
+
+fn deserialize_baudrate<'de, D>(deserializer: D) -> Result<Option<Option<NonZeroU32>>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    Ok(match Option::<i32>::deserialize(deserializer)? {
+        None => None,
+        Some(-1) => Some(None),
+        Some(baudrate) => Some(Some(NonZeroU32::new(baudrate as _).ok_or_else(|| {
+            serde::de::Error::custom(format!("invalid baudrate: {baudrate}"))
+        })?)),
+    })
+}
+
+impl RavedudeConfig {
+    pub fn from_args(args: &crate::Args) -> anyhow::Result<Self> {
+        Ok(Self {
+            general_options: RavedudeGeneralConfig {
+                open_console: args.open_console.then_some(true),
+                serial_baudrate: match args.baudrate {
+                    Some(serial_baudrate) => Some(
+                        NonZeroU32::new(serial_baudrate)
+                            .ok_or_else(|| anyhow::anyhow!("baudrate must not be 0"))?,
+                    ),
+                    None => None,
+                },
+                port: args.port.clone(),
+                reset_delay: args.reset_delay,
+                board: args.board.clone(),
+            },
+            board_config: Default::default(),
+        })
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug, Default)]
+#[serde(rename_all = "kebab-case")]
+pub struct RavedudeGeneralConfig {
+    pub open_console: Option<bool>,
+    pub serial_baudrate: Option<NonZeroU32>,
+    pub port: Option<std::path::PathBuf>,
+    pub reset_delay: Option<u64>,
+    pub board: Option<String>,
```

Comment:
Instead of making all fields `Option<>`s, you can also leverage [serde's default attribute](https://serde.rs/field-attrs.html#default).  Where it makes sense, of course.

Here, I think `open_console` would make much more sense that way:

```rust
#[serde(default)]
pub open_console: bool,
```

Reply:
Done! The only other place where I think this would be sensible is `do_chip_erase` in `BoardAvrdudeOptions`, but I don't want to use `#[serde(default)]` on it in case it's missing from the included `boards.toml` and goes unnoticed.